### ISSUE TITLE
Continue Server: Bad CPU type in executable

### ISSUE THREAD

**Issue by harperreed** [continuedev/continue#397]
**Describe the bug**
Continue installs an executable that cannot run on an Mac Mini 2023 M2. I imagine it would run if rosetta was installed. 

**To Reproduce**
1. Start VS Code
2. Install Continue
3. Click on Continue extension
4. Get stuck here: 
![image](https://github.com/continuedev/continue/assets/18504/5b62baf1-a42c-43d7-a4dc-7f3f7d07a48c)

5. go to terminal and find server executable: `~/.vscode/extensions/continue.continue-0.0.321/server/exe/run`
6. execute that: 

```shell

harper@Wizard ~ [124]> ~/.vscode/extensions/continue.continue-0.0.321/server/exe/run                                                                  (base)
exec: Failed to execute process '/Users/harper/.vscode/extensions/continue.continue-0.0.321/server/exe/run': Bad CPU type in executable.


```

**Expected behavior**
I would expect the server to run. 

**Environment**
- Hardware: Mac Mini 2023 M2
- Operating System: MacOS 13.4.1
- Python Version: 3.10.10
- Continue Version: 0.0.321

**Console logs**
```
[Extension Host] Error starting server: Error: spawn Unknown system error -86
    at ChildProcess.spawn (node:internal/child_process:441:11)
    at spawn (node:child_process:752:9)
    at Timeout.spawnChild (/Users/harper/.vscode/extensions/continue.continue-0.0.321/out/extension.js:55316:53)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
workbench.desktop.main.js:sourcemap:129 [Extension Host] Error caught: Error: spawn Unknown system error -86.

Retrying attempt 269...
workbench.desktop.main.js:sourcemap:129 [Extension Host] Error starting server: Error: spawn Unknown system error -86
    at ChildProcess.spawn (node:internal/child_process:441:11)
    at spawn (node:child_process:752:9)
    at Timeout.spawnChild (/Users/harper/.vscode/extensions/continue.continue-0.0.321/out/extension.js:55316:53)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
workbench.desktop.main.js:sourcemap:129 [Extension Host] Error caught: Error: spawn Unknown system error -86.

Retrying attempt 270...
workbench.desktop.main.js:sourcemap:129 [Extension Host] Error starting server: Error: spawn Unknown system error -86
    at ChildProcess.spawn (node:internal/child_process:441:11)
    at spawn (node:child_process:752:9)
    at Timeout.spawnChild (/Users/harper/.vscode/extensions/continue.continue-0.0.321/out/extension.js:55316:53)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
workbench.desktop.main.js:sourcemap:129 [Extension Host] Error caught: Error: spawn Unknown system error -86.

Retrying attempt 271...
workbench.desktop.main.js:sourcemap:129 [Extension Host] Error starting server: Error: spawn Unknown system error -86
    at ChildProcess.spawn (node:internal/child_process:441:11)
    at spawn (node:child_process:752:9)
    at Timeout.spawnChild (/Users/harper/.vscode/extensions/continue.continue-0.0.321/out/extension.js:55316:53)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
workbench.desktop.main.js:sourcemap:129 [Extension Host] Error caught: Error: spawn Unknown system error -86.

Retrying attempt 272...
workbench.desktop.main.js:sourcemap:129 [Extension Host] Error starting server: Error: spawn Unknown system error -86
    at ChildProcess.spawn (node:internal/child_process:441:11)
    at spawn (node:child_process:752:9)
    at Timeout.spawnChild (/Users/harper/.vscode/extensions/continue.continue-0.0.321/out/extension.js:55316:53)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
workbench.desktop.main.js:sourcemap:129 [Extension Host] Error caught: Error: spawn Unknown system error -86.

Retrying attempt 273...
workbench.desktop.main.js:sourcemap:129 [Extension Host] Error starting server: Error: spawn Unknown system error -86
    at ChildProcess.spawn (node:internal/child_process:441:11)
    at spawn (node:child_process:752:9)
    at Timeout.spawnChild (/Users/harper/.vscode/extensions/continue.continue-0.0.321/out/extension.js:55316:53)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
workbench.desktop.main.js:sourcemap:129 [Extension Host] Error caught: Error: spawn Unknown system error -86.

Retrying attempt 274...
workbench.desktop.main.js:sourcemap:129 [Extension Host] Error starting server: Error: spawn Unknown system error -86
    at ChildProcess.spawn (node:internal/child_process:441:11)
    at spawn (node:child_process:752:9)
    at Timeout.spawnChild (/Users/harper/.vscode/extensions/continue.continue-0.0.321/out/extension.js:55316:53)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
workbench.desktop.main.js:sourcemap:129 [Extension Host] Error caught: Error: spawn Unknown system error -86.
```

**Additional context**
Rosetta is not installed on my mac.

**Comment by sestinj** [comment]
@harperreed I appreciate the digging you did here. We'd just discovered this error the other day in https://github.com/continuedev/continue/issues/372, and are trying to work around GitHub not yet supporting Apple Silicon runners.

The first thing to do is probably make sure that there is more instructive error output for the user-I'll make that happen today.

In general, is installing Rosetta to fix this error something you would have been willing to do had it been suggested by Continue?

**Comment by harperreed** [comment]
Unsurprisingly, pyinstaller requires local libs to be of the target-arch architecture. I was unable to build a apple silicon binary on an intel mac locally.

**Comment by sestinj** [comment]
All good to know. This confirms the few things I'd tried. For now might just have to set up a self-hosted runner.

**Comment by adhney** [comment]
I had the same issue, I've a `MacBook Air M1`, I made it work for now by starting the server locally as given in: https://continue.dev/docs/how-continue-works

<img width="670" alt="image" src="https://github.com/continuedev/continue/assets/96858173/b47734d3-f893-4651-8e0a-9295735000f9">

It's just that every time I start my mac I'll have start the server doing the same thing again and again, tbh who shuts down Mac

**Comment by sestinj** [comment]
@harperreed @adhney I've just shipped a version that will use a server built for m1 Mac, hoping this solves the problem - please lmk!

Nice work setting up the server, glad our instructions weren't too hard to follow.

...and not me lol, Macs never get to sleep :(

**Comment by adhney** [comment]
yes @sestinj now it works even when I do not start the server manually. thanks a lot! btw great extension even greater execution.
<img width="465" alt="image" src="https://github.com/continuedev/continue/assets/96858173/ae3a29e7-8f70-41e4-abcb-20e87a447714">

**Comment by sestinj** [comment]
@adhney Much appreciated lol : ) And glad to see Continue has matched well with this unique color theme. You're always welcome to shoot me a message on [Discord](https://discord.gg/NWtdYexhMs) if you have questions or feedback!

**Comment by sestinj** [comment]
@harperreed We're a week into shipping with mac binaries now, and have verified from others that this is resolved, so am going to close this issue. If something's still up, please re-open!
